mipengine/controller/api/error_handlers.py

```diff
@@ -6,12 +6,12 @@
 from mipengine.controller.algorithm_executor import (
     NodeUnresponsiveAlgorithmExecutionException,
 )
-from mipengine.controller.api.exceptions import BadRequest
-from mipengine.controller.api.exceptions import BadUserInput
+from mipengine.exceptions import BadRequest
```

Comment:
`BadRequest` should only be thrown from the validator. Please move it there.

Reply:
`BadRequest` was not in the validator in the first place. It was in `mipengine/controller/api/exceptions.py` from which the validator was importing it. This module is simply merged with `mipengine/node_exceptions.py`. Why should I move it to the validator now?

Reply:
Because before it was at least inside the `controller` module, now it's not. It shouldn't be used outside of the controller scope. Since it's the only api specific exception now, you can just add it in the module its used instead of keeping a module with only one exception inside.

Reply:
OK I get your point. It's actually an important point. If Node code raises this exception it will probably be uncaught, since it was created for the Controller scope, and the app will crash. I will "fix" this by burying it deep inside the `controller` package and pray that no one will ever use it outside Controller. 

However, this just shows how bad a mistake was to artificially create a separation between the services by simply deleting subpackages. We all have to remember not to import from the wrong subpackages which puts an extra mental burden. OK, we are a small team and it's kinda feasible. What if the team grows? Would we remember to inform everyone about the correct place to import stuff based on context? Especially when any IDE would offer to automatically import any non-private class from anywhere in the `mipengine` package. Unfortunately, the only kind of test able to catch these mistakes is production tests which are slow and probably run only in CI.

Still, the problem remains. What about the `BadUserInput` class? Don't we run into the same problem by exposing it in the shared "area" of our package? Wasn't that also part of the Controller API? If someone imports and raises it in Node code, would it be handled?

Maybe the solution is what I originally suggested. To create a new class, e.g. `InvalidAlgorithmInput` defined in the `algorithms` folder and used only by algorithm flows. Controller will then catch it and raise the appropriate `BadUserInput`. I think it's better than just reusing some class that happened to have a nice name, but was intended for other use.

Reply:
The `BadRequest` should be inside the validator since the validator is the only module that should throw such an exception, since it's the module validating the algorithm request (hence throwing a 400 status code). 

The `BadUserInput` should be in the `mipengine` package since it could be thrown from multiple places of the engine causing a 460 status code error.

I think the rest of your comments are already discussed, in the team meeting, and decided upon.